sharedbuffers/mapped_struct.py

```diff
@@ -1719,6 +1738,403 @@ def __str__(self):
+                return self._search_key(elem, dcode, offset, lo, hi, hi // 2, True) < hi
+
+        h2 = _stable_hash(elem)
+        while lo < hi:
+            mid = lo + ((hi - lo) >> 1)
+            val = self.objlist._c_getitem(mid, dcode, objlen, itemsize, offset, _struct, None)
+            h1 = _stable_hash(val)
+            if h1 == h2:
+                # Equal hashes - Look in both directions for a match, without binary search
+                if val == elem:
+                    return True
+                while mid > 0:
+                    val = self.objlist._c_getitem(mid, dcode, objlen, itemsize, offset, _struct, None)
+                    h1 = _stable_hash(val)
+                    if h1 != h2:
+                        break
+                    elif val == elem:
+                        return True
+                    mid -= 1
+
+                mid = lo + ((hi - lo) >> 1)
+                while mid < hi:
+                    val = self.objlist._c_getitem(mid, dcode, objlen, itemsize, offset, _struct, None)
+                    h1 = _stable_hash(val)
+                    if h1 != h2:
+                        break
+                    elif val == elem:
+                        return True
+                    mid += 1
+                return False
+            elif h1 < h2:
+                lo = mid + 1
+            else:
+                hi = mid
+        return False
+
+    @cython.locals(i=cython.Py_ssize_t, dcode=cython.char, objlen=cython.longlong,
+        itemsize=cython.uchar, offset=cython.Py_ssize_t)
+    def __iter__(self):
+        if self.objlist is not None:
+            dcode, objlen, itemsize, offset, _struct = self.objlist._metadata()
+            for i in xrange(len(self)):
+                yield self.objlist._c_getitem(i, dcode, objlen, itemsize, offset, _struct, None)
+        else:
+            if self.bitrep_lo:
+                for i in xrange(64):
+                    if self.bitrep_lo & (1 << i):
+                        yield i
+                    elif not self.bitrep_lo >> i:
+                        break
+
+            if self.bitrep_hi:
+                for i in xrange(64):
+                    if self.bitrep_hi & (1 << i):
+                        yield i + 64
+                    elif not self.bitrep_hi >> i:
+                        break
+
+    def union(self, *seqs):
+        return frozenset(self).union(*seqs)
+
+    def intersection(self, *seqs):
+        return frozenset(self).intersection(*seqs)
+
+    def difference(self, *seqs):
+        return frozenset(self).difference(*seqs)
+
+    def symmetric_difference(self, *seqs):
+        return frozenset(self).symmetric_difference(*seqs)
+
+    @cython.ccall
+    @cython.locals(i=cython.Py_ssize_t, xlen=cython.Py_ssize_t,
+        pfset='proxied_frozenset', dcode=cython.char,
+        objlen=cython.longlong, itemsize=cython.uchar, offset=cython.Py_ssize_t)
+    def _frozenset_eq(self, x):
+        if isinstance(x, proxied_frozenset):
```

Comment:
Shouldn't we use `if type(x) is proxied_frozenset` to make it faster?

Reply:
we could, but that wouldn't work for subclasses, I think.